commons/src/lib.rs

```diff
@@ -55,9 +55,11 @@ pub fn ensure_query_params(
         .map(|(k, _)| k)
         .collect();
 
-    // Make sure all mandatory parameters are present.
-    if !required_params.is_subset(&query_keys) {
-        return Err(GraphError::MissingParams);
+    // Make sure no mandatory parameters are missing.
+    let mut missing: Vec<String> = required_params.difference(&query_keys).cloned().collect();
+    if !missing.is_empty() {
+        missing.sort();
```

Comment:
Just double-checking my understanding here. This sort is only for cosmetic reasons, right?

Reply:
This makes the output deterministic, allowing to check the error message for the real comma-separated values. It is not in a hot path, so I think it doesn't hurt. Alternatively I can drop this and make the test a little less strict.